pkgs/tools/games/minecraft/hmcl/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib, stdenv
+, fetchurl
+, jre
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hmcl";
+  version = "3.5.3.223";
+
+  src = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  icon = fetchurl {
+    url = "https://aur.archlinux.org/cgit/aur.git/plain/craft_table.png?h=hmcl-bin";
+    sha256 = "sha256-KYmhtTAbjHua/a5Wlsak5SRq+i1PHz09rVwZLwNqm0w";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+
+  desktopItems = lib.toList (makeDesktopItem {
+    name = "hmcl";
+    exec = "hmcl";
+    icon = "hmcl";
+    comment = "A Minecraft Launcher which is multi-functional, cross-platform and popular";
+    desktopName = "HMCL";
+    categories = [ "Game" ];
+  });
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/java/hmcl
+    install -Dm644 $src $out/share/java/hmcl/hmcl.jar
+    install -Dm644 $icon $out/share/icons/hicolor/48x48/apps/hmcl.png
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    makeWrapper ${jre}/bin/java $out/bin/hmcl \
```

Comment:
Any reason this isn't done in `installPhase`? Seems to me like there's no need to separate it.